### PR TITLE
Don't crash when WindowCenter or WindowWidth contain "NaN"

### LINKED ISSUE #1756: DICOM files where numeric tags contain "NaN" (don't ask) cause exceptions
**What is, according to you, missing from fo-dicom? Please describe.**
It would be nice if Fellow Oak DICOM could gracefully ignore these invalid values, as if they were empty or missing, instead of crashing loudly.

**Describe the solution you'd like**
If any of the following tags contain NaN or some other unparsable gobbledygook, pretend that they are empty instead:
Window Center (0028,1050)
Window width (0028,1051)
More specifically, in GrayScaleRenderOptions, we can deduce the window level values based on the pixel data instead of crashing while trying to parse the window level width and center. 

**Describe possible alternatives or existing workarounds you've considered**
A possible workaround would be to inspect these tags before trying to render, and removing them if they cannot be parsed to a valid numeric value.

You might be wondering, where do these NaN values come from? I don't know. We're seeing these come in from a German hospital (I won't name them), that's all I know.

### PR DESCRIPTION
**Pull request by amoerie**
Fixes #1756

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Ensure that we can handle DICOM files where WindowCenter or WindowWidth contain something unparsable such as NaN (we actually saw this in real production data recently)

### CODE REVIEW

File: Tests/FO-DICOM.Tests/Bugs/GH1756.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        [Fact]
+        public void GrayScaleRenderOptionsFromWindowCenterAndWidthEqualToNaNShouldFallbackCorrectly()
+        {
+            // Arrange
+            var dicomFile = DicomFile.Open("./Test Data/CT-MONO2-16-ankle");
+            
+            // Set value of <dicomTagName> to NaN
+            dicomFile.Dataset.ValidateItems = false;
+            dicomFile.Dataset.AddOrUpdate(DicomTag.WindowWidth, "NaN");
+            dicomFile.Dataset.AddOrUpdate(DicomTag.WindowCenter, "NaN");
```
Comment (mrbean-bremen) [inline Tests/FO-DICOM.Tests/Bugs/GH1756.cs]: Maybe also test invalid numbers (NaN is not invalid), and the other special values (positive/negative infinity).
Also, are there other tags (like Rescale Slope/Intercept) used that could have the same problem? (I haven't checked...)

Comment (amoerie) [reply Tests/FO-DICOM.Tests/Bugs/GH1756.cs]: NaN doesn't have any meaning in C#, so it might as well have said "Chocolate Donuts". :-) 
As for other special values, the only ones I can come up with are values <1 for window width, which is forbidden according to the standard.
Otherwise, I think these values can actually swing to wildly small or large values, fractional values included.

There may be similar issues with Rescale Slope / Intercept, and I initially planned to tackle those too, but I think that would drastically increase the scope of this PR, so I refrained from doing so.

File: FO-DICOM.Core/Imaging/GrayscaleRenderOptions.cs [hunk 2]
```diff
@@ -126,13 +126,14 @@ public static GrayscaleRenderOptions FromDataset(DicomDataset dataset, int frame
...
             if (dataset.TryGetValue(DicomTag.WindowWidth, 0, out double windowWidth) && windowWidth >= 1.0
-                && dataset.Contains(DicomTag.WindowCenter))
+                && dataset.TryGetValue(DicomTag.WindowCenter, 0, out double _))
             {
                 // If dataset contains WindowWidth and WindowCenter valid attributes used initially for the grayscale options
                 grayscaleRenderOptions = FromWindowLevel(dataset, frame);
             }
-            else if (dataset.FunctionalGroupValues(frame).TryGetValue(DicomTag.WindowWidth, 0, out double functionalWindowWidth) && functionalWindowWidth >= 1.0
-                && dataset.FunctionalGroupValues(frame).Contains(DicomTag.WindowCenter))
+            else if (dataset.FunctionalGroupValues(frame) is { } functionalGroupValues
```
Comment (amoerie) [inline FO-DICOM.Core/Imaging/GrayscaleRenderOptions.cs:134]: I'm actually a little proud of this little trick with pattern matching that lets us avoid calculating the FunctionalGroupValues three times. I say three times, because inside the "FromFunctionalWindowLevel" method it's still calculated another time,

But hey, progress!

**Comment by amoerie** [comment]
Failing pipeline caused by #1760

File: Tests/FO-DICOM.Tests/Bugs/GH1756.cs (hunk 1 above)
Comment (mrbean-bremen) [reply Tests/FO-DICOM.Tests/Bugs/GH1756.cs]: Ah sorry - there was recently a PR that did JSON parsing of float/double values, and I have somehow mixed these in my head. You are right, of course!

**Review by mrbean-bremen — APPROVED** [review]

**Review by gofal — APPROVED** [review]
